src/test.rs

```diff
@@ -136,7 +121,11 @@ mod marlin {
                 circ,
                 zk,
                 if zk { Some(rng) } else { None }
-            ).unwrap();
+            );
+
+            assert!(proof.is_ok());
+
+            let proof = proof.unwrap();
```

Comment:
Can we add a test case in which we pass wrong c and d with respect to the values of a and b and we assert the prover to return an Error ? Thanks !

Reply:
Done